packages/components/src/draggable/index.native.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * External dependencies
+ */
+import { Gesture, GestureDetector } from 'react-native-gesture-handler';
+import Animated, { useSharedValue } from 'react-native-reanimated';
+
+/**
+ * Draggable component
+ *
+ * @param {Object}                                      props                       Component props.
+ * @param {JSX.Element}                                 props.children              Children to be rendered.
+ * @param {?number}                                     props.maxDistance           Maximum distance, that defines how far the finger is allowed to travel during a long press gesture.
+ * @param {?number}                                     props.minDuration           Minimum time, that a finger must remain pressed on the corresponding view.
+ * @param {Function}                                    props.onDragEnd             Callback when dragging happens over an element.
+ * @param {Function}                                    props.onDragOver            Callback when dragging ends.
+ * @param {Function}                                    props.onDragStart           Callback when dragging stars.
+ * @param {import('react-native-reanimated').StyleProp} props.wrapperAnimatedStyles Animated styles for the wrapper component.
+ *
+ * @return {JSX.Element} The component to be rendered.
+ */
+export default function Draggable( {
+	children,
+	maxDistance = 1000,
+	minDuration = 500,
+	onDragEnd,
+	onDragOver,
+	onDragStart,
+	wrapperAnimatedStyles,
+} ) {
+	const isDragging = useSharedValue( false );
+	const hasPanStarted = useSharedValue( false );
+
+	const longPressGesture = Gesture.LongPress()
+		.onStart( ( ev ) => {
+			'worklet';
+			isDragging.value = true;
+
+			if ( onDragStart ) {
+				onDragStart( ev );
+			}
+		} )
+		.onEnd( () => {
+			if ( ! hasPanStarted.value && isDragging.value ) {
```

Comment:
If we remove it, `onDragEnd` gets called two times if both Gestures are active, since I started with only `LongPress` and then introduced `Pan` and then it changed to manually activating the `Pan` gesture, I think we could remove the `onEnd` from the `Pan` gesture and only have it for the `LongPress` gesture. With this we could also remove the `hasPanStarted` shared value to simplify things. What do you think?

Reply:
Good point, I've just checked that both `onEnd` callbacks are called when releasing the touch. In that case, I agree with what you suggested about calling `onDragEnd` only in the `LongPress` gesture 👍 .